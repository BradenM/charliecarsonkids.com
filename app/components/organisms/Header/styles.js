/**
 *
 * Header Styled Components
 *
 * Organism
 */

import Row from 'react-bootstrap/Row';
import styled from 'styled-components';

export const NavRow = styled(Row)`
  border-bottom: 3px solid #ecece1;
`;
