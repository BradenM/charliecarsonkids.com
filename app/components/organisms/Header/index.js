/**
 *
 * Header
 *
 * Organism
 */

import IconButton from 'atoms/IconButton';
import PlainLink from 'atoms/PlainLink';
import LogoBlack from 'images/logo_black.png';
import IconGroup from 'molecules/IconGroup';
import Navigation from 'molecules/Navigation';
import SocialBar from 'organisms/SocialBar';
import React from 'react';
import Col from 'react-bootstrap/Col';
import Container from 'react-bootstrap/Container';
import Image from 'react-bootstrap/Image';
import Row from 'react-bootstrap/Row';
import { useHistory } from 'react-router-dom';
import * as S from './styles';

const Header = () => {
  const history = useHistory();
  return (
    <Container fluid>
      <Row className="pt-2 px-3 header--container">
        <Col className="d-flex justify-content-start align-items-center ml-3">
          <SocialBar />
        </Col>
        <Col
          xs={3}
          xl={2}
          className="d-flex justify-content-center align-items-center"
        >
          <PlainLink to="/">
            <Image src={LogoBlack} fluid />
          </PlainLink>
        </Col>
        <Col className="d-flex justify-content-end align-items-center">
          <IconGroup>
            <IconButton
              onClick={() => history.push('/account/login')}
              name="user"
              iconSize="50"
              width="32"
            />
            <IconButton name="shopping" iconSize="50" width="32" />
          </IconGroup>
        </Col>
      </Row>
      <S.NavRow className="mb-3">
        <Col>
          <Navigation />
        </Col>
      </S.NavRow>
    </Container>
  );
};

Header.propTypes = {};

export default Header;
