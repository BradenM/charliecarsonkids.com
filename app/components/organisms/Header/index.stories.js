/**
 *
 * Stories for Header
 *
 * Organism
 */

import React from 'react';
import { host } from 'storybook-host';
import StoryRouter from 'storybook-react-router';
import Header from '.';

export default {
  title: 'Design|Organisms/Header',
  component: Header,
  decorators: [
    StoryRouter(),
    host({
      title: 'Header Component',
      align: 'center bottom',
      maxWidth: '100%',
      height: '100%',
      border: true,
    }),
  ],
  parameters: {
    centered: {
      disable: true,
    },
  },
};

export const defaultHeader = () => <Header />;
