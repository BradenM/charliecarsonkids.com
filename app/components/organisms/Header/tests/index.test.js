/**
 *
 * Tests for Header
 *
 * Organism
 */

import { render } from '@testing-library/react';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';
import Header from 'organisms/Header';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from '../../../../configureStore';

const renderComponent = (store, history, props = {}) =>
  render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Header {...props} />
      </ConnectedRouter>
    </Provider>,
  );

describe('<Header />', () => {
  let store;
  let history;

  beforeAll(() => {
    history = createMemoryHistory();
    store = configureStore({}, history);
  });
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderComponent(store, history);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderComponent(store, history);
    expect(firstChild).toMatchSnapshot();
  });
});
