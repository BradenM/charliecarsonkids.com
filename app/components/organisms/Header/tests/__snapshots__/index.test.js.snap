// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render and match the snapshot 1`] = `
<div
  class="container-fluid"
>
  <div
    class="pt-2 px-3 header--container row"
  >
    <div
      class="d-flex justify-content-start align-items-center ml-3 col"
    >
      <a
        class="A-ml9y9l-0 Navigation__StyledUnderlineLink-sc-11o2ur6-2 nbXcW navlink p my-0 active nav-link"
        color="hsl(345.70000000000005, 100%, 77.3%)"
        href="/"
      >
        Shop CCK
      </a>
    </div>
    <div
      class="d-flex justify-content-center align-items-center col-xl-2 col-3"
    >
      <img
        class="img-fluid"
        src="IMAGE_MOCK"
      />
    </div>
    <div
      class="d-flex justify-content-end align-items-center col"
    >
      <div
        class="border-0  list-group list-group-horizontal"
      >
        <div
          class="IconGroup__BorderlessItem-sc-1br5l8c-0 jUFpId list-group-item"
        >
          <div
            class="IconButton__IconContainer-pjjsvs-0 bKDZXR"
          >
            <div
              class="Anicon__HoverContainer-sc-119yqeq-0 iityHb anicon-container"
            >
              <div
                style="width: 30px; height: 100%; transform: rotate(90deg); cursor: pointer;"
              />
            </div>
          </div>
        </div>
        <div
          class="IconGroup__BorderlessItem-sc-1br5l8c-0 jUFpId list-group-item"
        >
          <div
            class="IconButton__IconContainer-pjjsvs-0 bKDZXR"
          >
            <div
              class="Anicon__HoverContainer-sc-119yqeq-0 iityHb anicon-container"
            >
              <div
                style="width: 30px; height: 100%; transform: rotate(0deg); cursor: pointer;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <nav
        class="Navigation__StyledNavbar-sc-11o2ur6-0 hoaiWk justify-content-center navbar navbar-expand-lg navbar-light"
      >
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-btn-container justify-content-center navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="justify-content-center navbar-nav"
          >
            <a
              class="A-ml9y9l-0 Navigation__StyledLink-sc-11o2ur6-1 fUswUo navlink p my-0 nav-link"
              color="hsl(345.70000000000005, 100%, 77.3%)"
              data-rb-event-key="new"
              href="new"
            >
              new
            </a>
            <a
              class="A-ml9y9l-0 Navigation__StyledLink-sc-11o2ur6-1 fUswUo navlink p my-0 nav-link"
              color="hsl(345.70000000000005, 100%, 77.3%)"
              data-rb-event-key="boys"
              href="boys"
            >
              boys
            </a>
            <a
              class="A-ml9y9l-0 Navigation__StyledLink-sc-11o2ur6-1 fUswUo navlink p my-0 nav-link"
              color="hsl(345.70000000000005, 100%, 77.3%)"
              data-rb-event-key="girls"
              href="girls"
            >
              girls
            </a>
            <a
              class="A-ml9y9l-0 Navigation__StyledLink-sc-11o2ur6-1 fUswUo navlink p my-0 nav-link"
              color="hsl(345.70000000000005, 100%, 77.3%)"
              data-rb-event-key="baby"
              href="baby"
            >
              baby
            </a>
            <a
              class="A-ml9y9l-0 Navigation__StyledLink-sc-11o2ur6-1 fUswUo navlink p my-0 nav-link"
              color="hsl(345.70000000000005, 100%, 77.3%)"
              data-rb-event-key="collections"
              href="collections"
            >
              collections
            </a>
            <a
              class="A-ml9y9l-0 Navigation__StyledLink-sc-11o2ur6-1 fUswUo navlink p my-0 nav-link"
              color="hsl(345.70000000000005, 100%, 77.3%)"
              data-rb-event-key="about"
              href="about"
            >
              about
            </a>
          </div>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
